### ISSUE TITLE
New-BcContainer error when including "-includePerformanceToolkit" parameter

### ISSUE THREAD

**Issue by BW-PA** [microsoft/navcontainerhelper#2187]
**Describe the issue**
Creating a container using New-BcContainer and including -includePerformanceToolkit causes the below error.  If I exclude the -includePerformanceToolkit parameter the container is created successfully.

```
Uninstalling Performance Toolkit from tenant default
WARNING: This license is not compatible with this version of Business Central.
WARNING: This license is not compatible with this version of Business Central.
Unpublishing Performance Toolkit
WARNING: This license is not compatible with this version of Business Central.
The Extension cannot be unpublished because it is required by the following apps: Performance Toolkit Samples by Microsoft 19.0.29894.32262, Performance Toolkit Tests by Microsoft 19.0.29894.32262
at <ScriptBlock>, <No file>: line 41
at Invoke-ScriptInBcContainer, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.18\ContainerHandling\Invoke-ScriptInNavContainer.ps1: line 43
at UnPublish-BcContainerApp, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.18\AppHandling\UnPublish-NavContainerApp.ps1: line 52
at Import-TestToolkitToBcContainer, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.18\ObjectHandling\Import-TestToolkitToNavContainer.ps1: line 159
at New-BcContainer, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.18\ContainerHandling\New-NavContainer.ps1: line 1937
at <ScriptBlock>, <No file>: line 9
at <ScriptBlock>, <No file>: line 2
The Extension cannot be unpublished because it is required by the following apps: Performance Toolkit Samples by Microsoft 19.0.29894.32262, Performance Toolkit Tests by Microsoft 19.0.29894.32262
At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.18\ContainerHandling\Invoke-ScriptInNavContainer.ps1:44 char:13
+             throw $_.Exception.Message
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (The Extension c...9.0.29894.32262:String) [], RuntimeException
    + FullyQualifiedErrorId : The Extension cannot be unpublished because it is required by the following apps: Performance Toolkit Samples by Microsoft 19.0.29894.32262, Performance Toolkit Tests by Microsoft 19.0.29894. 
   32262
```
 

**Scripts used to create container and cause the issue**
```
#Testing Generic BC Container script
    $containerName = 'basiccontainer'
    $password = 'admin'
    $securePassword = ConvertTo-SecureString -String $password -AsPlainText -Force
    $credential = New-Object pscredential 'admin', $securePassword
    $auth = 'UserPassword'
    $artifactUrl = Get-BcArtifactUrl -type 'Sandbox' -country 'au' -select 'Latest'
    New-BcContainer `
        -accept_eula `
        -containerName $containerName `
        -credential $credential `
        -auth $auth `
        -artifactUrl $artifactUrl `
        -imageName 'mybcimagev19' `
        -includeTestToolkit `
        -includePerformanceToolkit `
        -includeTestLibrariesOnly `
        -memoryLimit 8G `
        -updateHosts
```

**Full output of scripts**
```
BcContainerHelper is version 2.0.18
BcContainerHelper is running as administrator
Host is Microsoft Windows 10 Enterprise - 21H1
Docker Client Version is 20.10.10
Docker Server Version is 20.10.10
Removing container basiccontainer
Removing basiccontainer from container hosts file
Removing basiccontainer-* from container hosts file
Removing C:\ProgramData\BcContainerHelper\Extensions\basiccontainer
Fetching all docker images
Fetching all docker volumes
ArtifactUrl and ImageName specified
INFO: Windows 10 21H1/21H2 images are not yet available, using 2004 as these are found to work better than 20H2 on 21H1/21H2
WARNING: Unable to find matching generic image for your host OS. Using mcr.microsoft.com/businesscentral:10.0.19041.1348
Image mybcimagev19:sandbox-19.0.29894.32262-au-mt already exists
Using image mybcimagev19:sandbox-19.0.29894.32262-au-mt
Creating Container basiccontainer
Style: sandbox
Multitenant: Yes
Version: 19.0.29894.32262
Platform: 19.0.29884.32206
Generic Tag: 1.0.1.8
Container OS Version: 10.0.19041.1348 (2004)
Host OS Version: 10.0.19043.1348 (21H1)
WARNING: Host OS is 21H1 and Container OS is 2004, defaulting to process isolation. If you experience problems, add -isolation hyperv.
Using process isolation
Using locale en-AU
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Files in C:\ProgramData\BcContainerHelper\Extensions\basiccontainer\my:
- AdditionalOutput.ps1
- MainLoop.ps1
- SetupVariables.ps1
- updatehosts.ps1
Creating container basiccontainer from image mybcimagev19:sandbox-19.0.29894.32262-au-mt
1f6f99536907f1b7f96534171ead405b40071e5089dddb4fd9e69b53156a85b7
Waiting for container basiccontainer to be ready
Initializing...
Setting host.docker.internal to 10.1.10.65 in container hosts file (copy from host hosts file)
Setting gateway.docker.internal to 10.1.10.65 in container hosts file (copy from host hosts file)
Setting kubernetes.docker.internal to 127.0.0.1 in container hosts file (copy from host hosts file)
Setting host.containerhelper.internal to 172.17.112.1 in container hosts file
Starting Container
Hostname is basiccontainer
PublicDnsName is basiccontainer
Using NavUserPassword Authentication
Starting Local SQL Server
Starting Internet Information Server
Creating Self Signed Certificate
Self Signed Certificate Thumbprint 5F65C849E80378288CDE1BA1B3C2332D21DF52AF
Modifying Service Tier Config File with Instance Specific Settings
Starting Service Tier
Registering event sources
Creating DotNetCore Web Server Instance
Enabling Financials User Experience
Dismounting Tenant
Mounting Tenant
Mounting Database for default on server localhost\SQLEXPRESS with AllowAppDatabaseWrite = False
Sync'ing Tenant
Tenant is Operational
Creating http download site
Setting SA Password and enabling SA
Creating admin as SQL User and add to sysadmin
WARNING: This license is not compatible with this version of Business Central.
Creating SUPER user
WARNING: The password that you entered does not meet the minimum requirements. 
It should be at least 8 characters long and contain at least one uppercase 
letter, one lowercase letter, and one number.
WARNING: This license is not compatible with this version of Business Central.
WARNING: This license is not compatible with this version of Business Central.
Container IP Address: 172.17.126.130
Container Hostname  : basiccontainer
Container Dns Name  : basiccontainer
Web Client          : http://basiccontainer/BC/?tenant=default
Dev. Server         : http://basiccontainer
Dev. ServerInstance : BC
Dev. Server Tenant  : default
Setting basiccontainer to 172.17.126.130 in host hosts file
Setting basiccontainer-default to 172.17.126.130 in host hosts file
Setting basiccontainer-default to 172.17.126.130 in container hosts file

Files:
http://basiccontainer:8080/ALLanguage.vsix

Container Total Physical Memory is 31.9Gb
Container Free Physical Memory is 16.1Gb

Initialization took 103 seconds
Ready for connections!
Reading CustomSettings.config from basiccontainer
Creating Desktop Shortcuts for basiccontainer
Uninstalling Performance Toolkit from tenant default
WARNING: This license is not compatible with this version of Business Central.
WARNING: This license is not compatible with this version of Business Central.
Unpublishing Performance Toolkit
WARNING: This license is not compatible with this version of Business Central.
The Extension cannot be unpublished because it is required by the following apps: Performance Toolkit Samples by Microsoft 19.0.29894.32262, Performance Toolkit Tests by Microsoft 19.0.29894.32262
at <ScriptBlock>, <No file>: line 41
at Invoke-ScriptInBcContainer, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.18\ContainerHandling\Invoke-ScriptInNavContainer.ps1: line 43
at UnPublish-BcContainerApp, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.18\AppHandling\UnPublish-NavContainerApp.ps1: line 52
at Import-TestToolkitToBcContainer, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.18\ObjectHandling\Import-TestToolkitToNavContainer.ps1: line 159
at New-BcContainer, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.18\ContainerHandling\New-NavContainer.ps1: line 1937
at <ScriptBlock>, <No file>: line 9
at <ScriptBlock>, <No file>: line 2
The Extension cannot be unpublished because it is required by the following apps: Performance Toolkit Samples by Microsoft 19.0.29894.32262, Performance Toolkit Tests by Microsoft 19.0.29894.32262
At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.18\ContainerHandling\Invoke-ScriptInNavContainer.ps1:44 char:13
+             throw $_.Exception.Message
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (The Extension c...9.0.29894.32262:String) [], RuntimeException
    + FullyQualifiedErrorId : The Extension cannot be unpublished because it is required by the following apps: Performance Toolkit Samples by Microsoft 19.0.29894.32262, Performance Toolkit Tests by Microsoft 19.0.29894. 
   32262
...
```

**Additional context**
- does it happen all the time? _Yes, I can consistently recreate the issue_
- did it use to work? _Yes, I have a previous script that I saved for creating containers that includes the -includePerformanceToolkit parameter and it used to work, but now does not._

**Comment by freddydk** [comment]
This only happens when specifying an imagename (for image caching).
Fix included in the next containerhelper preview.

**Comment by BW-PA** [comment]
Thanks Freddy

**Comment by freddydk** [comment]
shipped in 2.0.23
